### PR TITLE
Add reducer and actions

### PR DESCRIPTION
**Pull request by Nasirkhan294**
### Add Reducer and Actions features:

- [ ] Added Redux Toolkit using (npm install react-redux @reduxjs/toolkit).
- [ ] Created  a Redux directory along with two child directories (books, categories)
- [ ] Configured a Redux store

### CODE REVIEW

**Comment by prolajumokeoni** [comment]
Hi @Nasirkhan294, you are missing future features,  you can update your future features section to have future features like styling👍🏿

**Comment by Nasirkhan294** [comment]
Hi @prolajumokeoni, noted with thanks!

File: src/redux/store.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import { configureStore } from '@reduxjs/toolkit';
+import booksReducer from './books/booksSlice';
+
+export default configureStore({
+  reducer: {
+    books: booksReducer,
```
Comment (prolajumokeoni) [inline src/redux/store.js:7]: - [ ] Kindly import the CategoriesReducer and the categories should be added to configureStore method 👍🏿

File: src/redux/store.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import { configureStore } from '@reduxjs/toolkit';
+import booksReducer from './books/booksSlice';
+
+export default configureStore({
```
Comment (prolajumokeoni) [inline src/redux/store.js]: - [ ] Kindly create a store `const store = configureStore({` and `export default store` instead of configureStore 👍🏿

**Review by prolajumokeoni — CHANGES_REQUESTED** [review]
Hi @Nasirkhan294 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
 - Installed and used  Redux Toolkit 👍🏿 



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Shedrack-Sunday — APPROVED** [review]
Hi @Nasirkhan294     👋

Congratulations! 🎉 Your project is complete and ready to be merged! 🚀

image


## Highlights
- Good job with the use of GitHub pages ✔️ 
- Kudos to meet the suggested changes by the previous reviwer    👍 

## Optional Suggestions 💡
While these comments with the [OPTIONAL] prefix won't prevent the approval of this pull request, I strongly recommend taking them into account as they can improve your code. Some of them were missed by the previous reviewer and addressing them will really enhance your application.
Cheers, and Happy coding!👏👏👏

If you have any questions or need clarification on anything, feel free to leave a comment in the pull request thread, and don't forget to tag me so I'll receive the notification.

As per the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561), you have a limited number of reviews per project. If you feel that the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**